php/tests/map_field_test.php

```diff
@@ -448,6 +448,24 @@ public function testMessage() {
 
     public function testMapElementIsReference()
     {
+        // Bool elements
```

Comment:
As a small note, tests like this may be easier to manage using a data provider, like this:

```php
public function mapElementIsReference()
{
    return [
        [
            [true => true],
            'setMapBoolBool'
        ], [
            [1 => 1],
            'setMapInt32Int32'
        ], [
            [1 => 1.0],
            'setMapInt32Double'
        ], [
            ['a' => 'a'],
            'setMapStringString'
        ], [
            [1 => new Sub()],
            'setMapInt32Message'
        ]
    ];
}

/**
 * @dataProvider mapElementIsReference
 */
public function testMapElementIsReference($values, $method)
{
    array_walk($values, function ($value) {});
    $m = new TestMessage();
    call_user_func($m, $method, $values);
}
```

It looks perfectly fine as is, but I often find data providers allow for more concise tests against a set of inputs.

Reply:
Thanks for the suggestion. I will try it next time.